lib/platform/github/index.ts

```diff
@@ -239,14 +251,14 @@ export async function initRepo({
         throw new Error(REPOSITORY_FORKED);
       }
     }
-    if (res.body.full_name && res.body.full_name !== repository) {
+    if (repo?.nameWithOwner && repo?.nameWithOwner !== repository) {
```

Comment:
Should we really get this far if `repo` is undefined?

Reply:
Mostly, for tests backward compatibility

Reply:
Line 235 already uses `repo.fork` so is it possible that all these other `repo?.` checks are unnecessary?